core/src/test/java/org/apache/struts2/interceptor/FetchMetadataInterceptorTest.java

```diff
@@ -6,27 +6,24 @@
 import com.opensymphony.xwork2.mock.MockActionInvocation;
 import org.apache.struts2.ServletActionContext;
 import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
 
-import javax.servlet.http.HttpServletResponse;
-import javax.servlet.http.HttpServletResponseWrapper;
 import java.util.Arrays;
-import java.util.HashSet;
-
-// TODO Vary headers tests
-// TODO  integration test where we set up a few interceptors and check that the chain gets interrupted when an attack reaches our interceptor
 
 public class FetchMetadataInterceptorTest extends XWorkTestCase {
 
-    private final FetchMetadataInterceptor interceptor = new FetchMetadataInterceptor(new HashSet<>(Arrays.asList("/foo", "/bar")));
+    private final FetchMetadataInterceptor interceptor = new FetchMetadataInterceptor();
 
 
     public void testNoSite() throws Exception {
         MockActionInvocation mai = new MockActionInvocation();
         MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
```

Comment:
Let's remove code duplication in this file by converting the `mai`, `request` and `response` to fields and setting them up in the setUp method. The following lines can be written only once for all tests:

```      MockActionInvocation mai = new MockActionInvocation();
        MockHttpServletRequest request = new MockHttpServletRequest();
        MockHttpServletResponse response = new MockHttpServletResponse();

        ServletActionContext.setRequest(request);
        ServletActionContext.setResponse(response);

        ActionContext context = ServletActionContext.getActionContext();
        mai.setInvocationContext(context);
```

And remove this setup code from all tests, please.